src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -53,20 +53,48 @@ private static bool IsEnabled()
                    s_diagnosticListener.IsEnabled();
         }
 
-        private static Activity? CreateActivity(HttpRequestMessage requestMessage)
+        private static Activity? StartActivity(HttpRequestMessage request)
         {
             Activity? activity = null;
             if (s_activitySource.HasListeners())
             {
-                activity = s_activitySource.CreateActivity(DiagnosticsHandlerLoggingStrings.ActivityName, ActivityKind.Client);
+                Uri? requestUri = request.RequestUri;
+                if (requestUri?.IsAbsoluteUri == false)
+                {
+                    requestUri = null;
+                }
+
+                // The following tags might be important for making sampling decisions and should be provided at span creation time.
+                // https://github.com/open-telemetry/semantic-conventions/blob/5077fd5ccf64e3ad0821866cc80d77bb24098ba2/docs/http/http-spans.md?plain=1#L213-L219
+                KeyValuePair<string, object?>[] tags = requestUri is not null
+                    ? new KeyValuePair<string, object?>[4]
+                    : new KeyValuePair<string, object?>[1];
```

Comment:
In what situation will we get to this point with a null requestUri?

Reply:
With the code above

```csharp
if (requestUri?.IsAbsoluteUri == false)
{
    requestUri = null;
}
```

This logic is covering two cases: (1) uri is null, (2) uri is relative. While `HttpClient` is guarding against these cases,  `HttpMessageInvoker` may pass such Uri-s to to DiagnosticsHandler and eventually to platform handlers.

While I see that BrowserHttpHandler can deal with relative Uri-s, IDK if is the null uri is a valid case. If we think it it isn't, we can add logic to `HttpClientHandler` that diasallows it in its `Send(Async)` implementation before passing it to any `HttpMessageHandlerStage`, but I would prefer to open a separate issue to discuss that instead of sneaking the change into this PR.